packages/transaction-controller/src/TransactionController.ts

```diff
@@ -975,34 +1248,38 @@ export class TransactionController extends BaseControllerV1<
       transactionMeta,
     );
 
-    const cancelTransactionMeta: TransactionMeta = {
```

Comment:
Hope this clears up the intention behind this!

- What we're gaining:

Type annotations prevent the user-supplied type from being narrowed further. 

In this case, the compiler is capable of inferring that `cancelTransactionMeta` does not fall under the `TransactionMeta` discriminated union branch that has an `error` property. However, the annotation is preventing the compiler from accurately narrowing `cancelTransactionMeta` to its true type. 

In other words, the annotation is artificially *widening* the type of `cancelTransactionMeta`, losing type information in the process. 

- What we're not losing:

If validation must take place in this line, adding the annotation is an option, but I'm not sure why that would be necessary here. 

Since this is a TypeScript codebase with little to no `any` usage, it is guaranteed that any call site where `cancelTransactionMeta` is consumed will have type checks for validating that it conforms to `TransactionMeta`.

Therefore, an "early exit" pattern to redundantly/preemptively execute this type check is unnecessary for a codebase with strong type-safety guarantees.

Reply:
@matthewwalsh0 To be fair, your concerns are not unwarranted at all -- validating a type at the source is still a valid use case. Thankfully, TypeScript added support for this on v4.9 with the new `satisfies` operator.

`satisfies` targets exactly this issue by letting us provide a wider, abstract type constraint for validation, while simultaneously allowing the compiler to narrow down to a more accurate type.

So we will have a better way to handle this very soon. Just wanted to add that bit of context.